libnopegl/src/pgcraft.c

```diff
@@ -1271,6 +1271,33 @@ static void probe_texture_infos(struct pgcraft *s)
     }
 }
 
+static void probe_ublocks(struct pgcraft *s)
+{
+    struct darray *array = &s->filtered_pipeline_info.desc.buffers;
+    const struct pipeline_resource_desc *buffer_descs = ngli_darray_data(array);
+    const size_t nb_buffer_descs = ngli_darray_count(array);
```

Comment:
can be in the loop; we do that everywhere

Reply:
Done.

Reply:
And move `const struct pipeline_resource_desc *buffer_descs = ngli_darray_data(array);` just before the loop